src/components/Instructions/InstructionItem.tsx

```diff
@@ -116,7 +116,7 @@ export const InstructionItem = ({
           </TableCell>
           <TableCell className="p-1.5">
             <span className="">
-              {"args" in programRow && mapInstructionsArgsByType(programRow.args, numeralSystem)}
+              {"args" in programRow && mapInstructionsArgsByType(programRow.args, numeralSystem, programRow.counter)}
```

Comment:
You may try to remove condition here. It's mostly a type check cause of wrong typing. With your newest changes maybe it's not necessary.

Reply:
the condition is necessary -`disassemblify` doesn't return `args` in case of error:
<img width="634" alt="Screenshot 2024-10-13 at 13 05 14" src="https://github.com/user-attachments/assets/c9be3d22-e289-4763-89e5-b63e48dc7fc3">